subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/FindAndroidAppTaskAction.kt

```diff
@@ -0,0 +1,35 @@
+package com.avito.module.internal.dependencies
+
+import com.avito.module.configurations.ConfigurationType
+import com.avito.module.internal.dependencies.FindAndroidAppTaskAction.Verdict.MultipleSuitableApps
+import com.avito.module.internal.dependencies.FindAndroidAppTaskAction.Verdict.NoSuitableApps
+import com.avito.module.internal.dependencies.FindAndroidAppTaskAction.Verdict.OneSuitableApp
+import org.gradle.api.Project
+import org.gradle.util.Path
+
+internal class FindAndroidAppTaskAction(
+    private val graphBuilder: DependenciesGraphBuilder
+) {
+
+    fun findAppFor(modules: List<Path>, configuration: ConfigurationType): Verdict {
+        val suitableApps = graphBuilder
+            .buildDependenciesGraphFlatten(configuration)
+            .filter { (_, dependencyProjects) ->
+                dependencyProjects
+                    .map { Path.path(it.path) }
+                    .toSet()
+                    .containsAll(modules)
```

Comment:
I think we have another common scenario - partial contains. 
In this case, I'd want to get advice about the most suitable app.

Reply:
I will create a task